backends/dpdk/dpdkArch.cpp

```diff
@@ -2440,7 +2444,26 @@ void CollectAddOnMissTable::postorder(const IR::MethodCallStatement *mcs) {
     }
     auto ctxt = findContext<IR::P4Action>();
     BUG_CHECK(ctxt != nullptr, "%1%: add_entry extern can only be used in an action", mcs);
-
+    auto it = structure->defActs.find(ctxt->name.name);
+    auto add_on_miss = it->second->properties->getProperty("add_on_miss");
+    if (add_on_miss == nullptr) return;
+    if (add_on_miss->value->is<IR::ExpressionValue>()) {
+        auto expr = add_on_miss->value->to<IR::ExpressionValue>()->expression;
+        if (!expr->is<IR::BoolLiteral>()) {
+            ::error(ErrorType::ERR_UNEXPECTED,
+                    "%1%: expected boolean for 'add_on_miss' property", add_on_miss);
+            return;
+        } else {
+            use_add_on_miss = expr->to<IR::BoolLiteral>()->value;
+        }
+    }
+    if (use_add_on_miss) {
+        if (it == structure->defActs.end()) {
+            ::error(ErrorType::ERR_UNEXPECTED, "%1% is not inside a miss action: %2%:",
```

Comment:
It seems good to have at least one test program that exercises detecting this error, and confirms that the desired error message is printed.

Reply:
An existing tests does test the code. See p4c/testdata/p4_16_samples/pna-add-on-miss-err.p4.test